xkafka/middleware/recover.go

```diff
@@ -0,0 +1,25 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gojekfarm/xtools/xkafka"
+)
+
+// RecoverMiddleware catches panics and prevents the request goroutine from crashing.
+func RecoverMiddleware() xkafka.MiddlewareFunc {
```

Comment:
Is this required with the introduction of `sourcegraph/conc` since it also [handles panics](https://github.com/sourcegraph/conc#goal-2-handle-panics-gracefully)?

Reply:
Might be still required for producers middleware